pcsx2/gui/ConsoleLogger.h

```diff
@@ -220,3 +223,29 @@ class ConsoleLogFrame : public wxFrame
 
 	void OnLoggingChanged();
 };
+
+template<typename ... Args>
+void OSDlog(ConsoleColors color, bool console, const std::string& format, Args ... args) {
+	if(!GSosdLog) return;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-security"
+#if defined(_MSC_VER) && _MSC_VER < 1900
+	size_t size = _snprintf( nullptr, 0, format.c_str(), args ... ) + 1; // Extra space for '\0'
+#else
+	size_t size = snprintf( nullptr, 0, format.c_str(), args ... ) + 1; // Extra space for '\0'
+#endif
+	std::vector<char> buf(size);
+#if defined(_MSC_VER) && _MSC_VER < 1900
+	_snprintf( buf.data(), size, format.c_str(), args ... );
+#else
+	snprintf( buf.data(), size, format.c_str(), args ... );
+#endif
+#pragma GCC diagnostic pop
+	GSosdLog(wxString( buf.data(), buf.data() + size - 1 ).utf8_str(), wxGetApp().GetProgramLog()->GetRGBA(color));
+
+	if(console) Console.WriteLn( color, format, std::forward<Args>(args)... );
```

Comment:
It should probably be `format.c_str()`. That should solve the garbage text issue that FlatOutPS2 described.